CorsixTH/CorsixTH
# [RFC] Add mouse capture option to option menus

This PR adds a new toggle to the options menu and in game tool-bar to toggle whether the mouse should be captured in the current window or not.

I've re-factored how the y positions of buttons in the options menu are calculated, instead of having to recalculate the Y position of all buttons when we add a new one. Instead we keep track internally of the y position and call a function which gives us a new co-ord and increments that counter.

I've adds three new strings which must be localised too:
- menu_options.capture_mouse
- options_window.capture_mouse
- tooltip.options_window.capture_mouse

I've also updated .gitignore with some additional files which should be ignored.

Fixes #1309 

----

To test:
- Start CorsixTH
- Go to options menu and check button is sane (e.g. is pressed on when the setting is on)
- Try toggling the option and check the you can move the mouse out of the window or cannot
- Check the configuration file is updated each time you click it
- Close and open the options window, is the state preserved
- Go into any level
- Go to the top toolbar and check that the state you left it in is preserved
- Check that if you toggle the option the mouse is correctly captured
- Check the config is saved each time you toggle it on or off
